File: common/mrtjp/projectred/transmission/TransmissionRecipes.java
package mrtjp.projectred.transmission;

import codechicken.microblock.ItemMicroPart;
import cpw.mods.fml.common.registry.GameRegistry;
import mrtjp.projectred.core.ItemPart.EnumPart;
import mrtjp.projectred.core.PRColors;
import mrtjp.projectred.core.ShapedOreNBTRecipe;
import mrtjp.projectred.core.ShapelessOreNBTRecipe;
import net.minecraft.block.Block;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraftforge.oredict.ShapedOreRecipe;
import net.minecraftforge.oredict.ShapelessOreRecipe;

public class TransmissionRecipes
{
    public static void initTransmissionRecipes()
    {
        initWireRecipes();
        initPartRecipes();
    }

    private static void initWireRecipes()
    {
        /** Red Alloy Wires **/
        GameRegistry.addRecipe(WireDef.RED_ALLOY().getItemStack(12),
                " r ",
                " r ",
                " r ",
                'r', EnumPart.REDINGOT.getItemStack()
                );

        /** Insulated Wires **/
        for (int i = 0; i < WireDef.INSULATED_WIRE().length; i++)
        {
            WireDef w = WireDef.INSULATED_WIRE()[i];
            GameRegistry.addRecipe(w.getItemStack(12),
                    "WrW",
                    "WrW",
                    "WrW",
                    'W', new ItemStack(Block.cloth, 1, PRColors.get(i).woolId()),
                    'r', EnumPart.REDINGOT.getItemStack()
                    );
        }
        for (WireDef w : WireDef.INSULATED_WIRE()) // Recoloring
            GameRegistry.addRecipe(new ShapelessOreRecipe(w.getItemStack(),
                    PRColors.get(w.meta()-WireDef.INSULATED_0().meta()).getOreDict(),
                    WireDef.oreDictDefinitionInsulated(),
                    Item.silk
                    ));
        for (WireDef w : WireDef.INSULATED_WIRE()) // Framed Recoloring
            GameRegistry.addRecipe(new ShapelessOreNBTRecipe(w.getFramedItemStack(),
                    PRColors.get(w.meta()-WireDef.INSULATED_0().meta()).getOreDict(),
                    WireDef.oreDictDefinitionInsFramed(),
                    ItemMicroPart.create(769, Block.wood.getUnlocalizedName())
                    ).setCheckNBT());


        /** Bundled Cables **/
        GameRegistry.addRecipe(new ShapedOreRecipe(WireDef.BUNDLED_N().getItemStack(),
                "SWS",
                "WWW",
                "SWS",
                'S', Item.silk,
                'W', WireDef.oreDictDefinitionInsulated()
                ));
        for (WireDef w : WireDef.BUNDLED_WIRE()) // recoloring
        {
            if (w == WireDef.BUNDLED_N())
                continue;
            GameRegistry.addRecipe(new ShapelessOreRecipe(w.getItemStack(),
                    PRColors.get(w.meta()-WireDef.BUNDLED_0().meta()).getOreDict(),
                    WireDef.oreDictDefinitionBundled(),
                    Item.silk
                    ));
        }

        /** Framed Wiring **/
        for (WireDef w : WireDef.VALID_WIRE())
            if (w.hasFramedForm())
                // Regular to framed
                GameRegistry.addRecipe(new ShapedOreNBTRecipe(w.getFramedItemStack(),
                        "sss",
                        "sis",
                        "sss",
                        'i', w.getItemStack(),
                        's', ItemMicroPart.create(769, Block.wood.getUnlocalizedName())
                        ).setCheckNBT());
    }

    private static void initPartRecipes()
    {
        /** Wired Plate **/
        GameRegistry.addRecipe(EnumPart.WIREDPLATE.getItemStack(),
                "r",
                "p",
                'r', WireDef.RED_ALLOY().getItemStack(),
                'p', EnumPart.PLATE.getItemStack()
                );

        /** Bundled Plate **/
        GameRegistry.addRecipe(EnumPart.BUNDLEDPLATE.getItemStack(),
                "r",
                "p",
                'r', WireDef.BUNDLED_N().getItemStack(),
                'p', EnumPart.PLATE.getItemStack()
                );
    }
}

File: src/mrtjp/projectred/transmission/TransmissionRecipes.scala
package mrtjp.projectred.transmission

import java.lang.{Character => JC}

import codechicken.lib.colour.EnumColour
import mrtjp.projectred.core.PartDefs
import net.minecraft.init.{Blocks, Items}
import net.minecraft.item.ItemStack
import net.minecraftforge.fml.common.registry.GameRegistry
import net.minecraftforge.oredict.{ShapedOreRecipe, ShapelessOreRecipe}

object TransmissionRecipes
{
    def initTransmissionRecipes()
    {
        WireDef.initOreDict()
        initWireRecipes()
        initPartRecipes()
    }

    private def initWireRecipes()
    {
        GameRegistry.addRecipe(new ShapedOreRecipe(WireDef.RED_ALLOY.makeStack(12),
            " r ", " r ", " r ",
            'r':Character, PartDefs.oreDictDefinitionRedIngot))

        //Insulated wire
        for (w <- WireDef.INSULATED_WIRES)
            GameRegistry.addRecipe(new ShapedOreRecipe(w.makeStack(12),
                "WrW", "WrW", "WrW",
                'W':Character, new ItemStack(Blocks.WOOL, 1, EnumColour.values()(w.meta-WireDef.INSULATED_0.meta).getWoolDamage),
                'r':Character, PartDefs.oreDictDefinitionRedIngot))

        //Insulated wire recolouring
        for (w <- WireDef.INSULATED_WIRES)
            GameRegistry.addRecipe(new ShapelessOreRecipe( w.makeStack,
                EnumColour.values()(w.meta-WireDef.INSULATED_0.meta).getOreDictionaryName,
                WireDef.oreDictDefinitionInsulated))

        //Framed insulated wire recolouring
        for (w <- WireDef.INSULATED_WIRES)
            GameRegistry.addRecipe(new ShapelessOreRecipe(w.makeFramedStack,
                EnumColour.values()(w.meta-WireDef.INSULATED_0.meta).getOreDictionaryName,
                WireDef.oreDictDefinitionInsFramed
            ))

        //Bundled cable
        GameRegistry.addRecipe(new ShapedOreRecipe(WireDef.BUNDLED_N.makeStack,
            "SWS", "WWW", "SWS",
            'S':Character, Items.STRING,
            'W':Character, WireDef.oreDictDefinitionInsulated))

        //Bundled cable recolouring
        for (w <- WireDef.BUNDLED_WIRES) if (w != WireDef.BUNDLED_N)
            GameRegistry.addRecipe(new ShapelessOreRecipe(w.makeStack, EnumColour.values()(w.meta-WireDef.BUNDLED_0.meta).getOreDictionaryName,
                WireDef.oreDictDefinitionBundled, Items.STRING))

        //Framed insulated+bundled wire
        for (w <- WireDef.values) if (w.hasFramedForm)
            GameRegistry.addRecipe(new ShapedOreRecipe(w.makeFramedStack,
                "sss", "sis", "sss",
                's':Character, "stickWood",
                'i':Character, w.makeStack
            ))

        GameRegistry.addRecipe(new ShapedOreRecipe(WireDef.POWER_LOWLOAD.makeStack(12),
            "bib","yiy", "bib",
            'b':JC, new ItemStack(Blocks.WOOL, 1, EnumColour.BLUE.getWoolDamage),
            'i':JC, "ingotElectrotineAlloy",
            'y':JC, new ItemStack(Blocks.WOOL, 1, EnumColour.YELLOW.getWoolDamage)
        ))
    }

    private def initPartRecipes()
    {
        GameRegistry.addRecipe(PartDefs.WIREDPLATE.makeStack,
            "r", "p",
            'r':Character, WireDef.RED_ALLOY.makeStack,
            'p':Character, PartDefs.PLATE.makeStack)

        GameRegistry.addRecipe(PartDefs.BUNDLEDPLATE.makeStack,
            "r", "p",
            'r':Character, WireDef.BUNDLED_N.makeStack,
            'p':Character, PartDefs.PLATE.makeStack)
    }
}